pkg/processor/runtime/rpc/abstract.go

```diff
@@ -203,15 +212,27 @@ func (r *AbstractRuntime) processEventAndWaitForResult(event nuclio.Event, funct
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to allocate connection for processing event")
 	}
-	return connectionInstance.ProcessEvent(event, functionLogger)
+	result, err := connectionInstance.ProcessEvent(event, functionLogger)
+	go func() {
+		// release connection after processing event
+		// in goroutine to avoid blocking the processing
+		r.connectionManager.Release(connectionInstance)
+	}()
```

Comment:
No need for an extra anonymous function
```suggestion
	// release connection after processing event
	// in goroutine to avoid blocking the processing
	go r.connectionManager.Release(connectionInstance)
```